src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/MarkJavaObjects.cs

```diff
@@ -196,7 +199,38 @@ void PreserveInvoker (TypeDefinition type)
 			PreserveConstructors (type, invoker);
 			PreserveIntPtrConstructor (invoker);
 			PreserveInterfaceMethods (type, invoker);
+#if ILLINK
+			PreserveInterfaces (invoker);
+#endif // ILLINK
+		}
+
+#if ILLINK
+		void PreserveInterfaces (TypeDefinition type)
+		{
+			if (!type.HasInterfaces)
+				return;
+
+			// Return if [Register(DoNotGenerateAcw=true)] is on the type
+			if (type.HasCustomAttributes) {
+				foreach (var attr in type.CustomAttributes) {
+					if (attr.AttributeType.FullName == "Android.Runtime.RegisterAttribute") {
```

Comment:
For "future-proofing", this should *also* check for [`Java.Interop.JniTypeSignatureAttribute.GenerateJavaPeer`](https://github.com/xamarin/java.interop/blob/fcc33ce2419c8f0e4ea41418ef8e07fcc4b81e2b/src/Java.Interop/Java.Interop/JniTypeSignatureAttribute.cs#L31).